napari/layers/transforms.py

```diff
@@ -41,14 +41,22 @@ def inverse(self):
             raise ValueError('Inverse function was not provided.')
 
 
-class TransformChain(Transform, ListModel):
+class TransformChain(ListModel, Transform):
     def __init__(self, transforms=[]):
-        super().__init__(basetype=Transform, iterable=transforms, lookup=id)
+        super().__init__(
+            basetype=Transform,
+            iterable=transforms,
+            lookup={str: lambda q, e: q == e.name},
+        )
 
     def __call__(self, coords):
         return tz.pipe(coords, *self)
 
+    def __newlike__(self, iterable):
+        return ListModel(self._basetype, iterable, self._lookup)
```

Comment:
@kne42 would you mind clarifying what these lines do? We copied them from LayerList but we don't really understand what's happening. (At least I don't!)

Reply:
__newlike__ is used when you want to create a new List with the same settings but out of a different iterable. it is used in slicing and copy

Reply:
Is it standard Python (I couldn't find any docs?) or your own invention? If the latter, the common practice is to use single-underscore wrapping, leaving dunder to standard Python protocols.